File: 125palind_sum.py
import math

def isPalin(x):
	rev = 0
	x1  = x
	while x1 > 0:
		rev = rev * 10 + x1 % 10
		x1 /= 10
	return rev == x


def findAll(limit):
	pals = set()
	def test(x):
		if isPalin(x):
			pals.add(x)
	l2 = int(math.sqrt(limit))+1
	for s in range(1, l2):
		ss = s * s
		#test(ss)
		for e in range(s+1, l2):
			ss += e * e
			if ss > limit:
				break
			test(ss)
	return pals
			
s = findAll(1e8)
print s
print len(s)
print sum(s)

File: 125palind_sum.c
#include "stdlib.h"
#include "stdio.h"

int isPalin(unsigned long x)
{
	unsigned long rev = 0;
	for (unsigned long x1 = x; x1 > 0; x1 /= 10)
	{
		rev = rev * 10 + x1 % 10;
	}
	return rev;
}

void findAll(int limit)
{
	unsigned long ss;
	for (int s = 0; s < limit; ++s)
	{
		ss = s * s;
		
		for (int e = s+1; e < limit; ++e)
		{

		}
	}
}

int main(int argc, char const *argv[])
{
}
